tests/prague/eip7692_eof_v1/eip7069_extcall/test_calls.py

```diff
@@ -644,3 +645,275 @@ def test_eof_calls_clear_return_buffer(
+    state_test: StateTestFiller,
+    pre: Alloc,
+    sender: EOA,
+):
+    """Test EOF contracts calls handle value calls with not enough balance"""
+    env = Environment()
+
+    noop_callee_address = pre.deploy_contract(Container.Code(Op.STOP))
+
+    calling_contract_address = pre.deploy_contract(
+        Container.Code(
+            Op.SSTORE(slot_call_result, Op.EXTCALL(address=noop_callee_address, value=1))
+            + Op.SSTORE(slot_code_worked, value_code_worked)
+            + Op.STOP
+        )
+    )
+    tx = Transaction(
+        sender=sender,
+        to=Address(calling_contract_address),
+        gas_limit=50000000,
+        data="",
+    )
+
+    calling_storage = {
+        slot_code_worked: value_code_worked,
+        slot_call_result: value_eof_call_reverted,
+    }
+
+    post = {
+        calling_contract_address: Account(storage=calling_storage),
+    }
+
+    state_test(
+        env=env,
+        pre=pre,
+        post=post,
+        tx=tx,
+    )
+
+
+@pytest.mark.parametrize(
+    "opcode",
+    [
+        Op.EXTCALL,
+        Op.EXTDELEGATECALL,
+        Op.EXTSTATICCALL,
+    ],
+)
+def test_eof_calls_msg_depth(
+    state_test: StateTestFiller,
+    pre: Alloc,
+    sender: EOA,
+    opcode: Op,
+):
+    """
+    Test EOF contracts calls handle msg depth limit correctly (1024).
+    NOTE: due to block gas limit and the 63/64th rule this limit is unlikely to be hit
+          on mainnet.
+    """
+    # Not a precise gas_limit formula, but enough to exclude risk of gas causing the failure.
+    gas_limit = int(200000 * (64 / 63) ** 1024)
+    env = Environment(gas_limit=gas_limit)
+
+    callee_address = Address(0x5000)
+
+    # Flow of the test:
+    # `callee_code` is recursively calling itself, passing msg depth as calldata
+    # (kept with the `MSTORE(0, ADD(...))`). When maximum msg depth is reached
+    # the call fails and starts returning. The deep-most frame returns:
+    #   - current reached msg depth (expected to be the maximum 1024), with the
+    #     `MSTORE(32, ADD(...))`
+    #   - the respective return code of the EXT*CALL (expected to be 1 - light failure), with the
+    #     `MSTORE(64, NOOP)`. Note the `NOOP` is just to appease the `Op.MSTORE` call, the return
+    #     code value is actually coming from the `Op.DUP1`
+    # When unwinding the msg call stack, the intermediate frames return whatever the deeper callee
+    # returned with the `RETURNDATACOPY` instruction.
+    returndatacopy_block = Op.RETURNDATACOPY(32, 0, 64) + Op.RETURN(32, 64)
+    deep_most_result_block = (
+        Op.MSTORE(32, Op.ADD(Op.CALLDATALOAD(0), 1)) + Op.MSTORE(64, Op.NOOP) + Op.RETURN(32, 64)
+    )
+    rjump_offset = len(returndatacopy_block)
+
+    callee_code = Container.Code(
+        # current stack depth in memory bytes 0-31
+        Op.MSTORE(0, Op.ADD(Op.CALLDATALOAD(0), 1))
+        # pass it along deeper as calldata
+        + opcode(address=callee_address, args_size=32)
```

Comment:
```suggestion
        + opcode(address=Op.ADDRESS, args_size=32)
```
We could use `Op.ADDRESS` here to avoid hard-coding the address into the pre-alloc.

The only caveat is that we need to use `Op.EXTCALL` in the `calling_contract_address` instead of `opcode` because the trick doesn't work when the top level call is initiated with `Op.EXTDELEGATECALL`, but I think the test still accomplishes its purpose with this change.